### PR TITLE
[pro#347] Batch authorities browsing

### PR DESCRIPTION
**Pull request by gbp**
Fixes https://github.com/mysociety/alaveteli-professional/issues/347
Fixes https://github.com/mysociety/alaveteli-professional/issues/402

Requires and closes ~~https://github.com/mysociety/alaveteli/pull/4450~~
Requires and closes ~~https://github.com/mysociety/alaveteli/pull/4457~~

Allow browsing of existing authorities categories/ tags in the batch builder. Includes 'Add all' buttons to select all authorities within a tag to the batch request.

### CODE REVIEW

File: config/general.yml-example [hunk 1]
```diff
@@ -1068,6 +1068,14 @@ PRO_CONTACT_NAME: 'Alaveteli Professional'
 # ---
 PRO_SITE_NAME: 'Alaveteli Professional'
 
+# The total number of authorities that can be added to a Alaveteli Professional
+# batch request.
+#
+# PRO_BATCH_LIMIT - Integer (default: 500)
+#
+# ---
+PRO_BATCH_LIMIT:
```
[the remark this answers is not in the document]
Comment (lizconlan) [reply config/general.yml-example]: also `PRO_BATCH_LIMIT` may be ambiguous as we're limiting the number of batches a user can make and the number of authorities we're allowing (hello from the middle of the impose a limit on the number of batches ticket!)

**Comment by gbp** [comment]
@zarino there seems to be some inconsistency in the layout and styling between browsers - see below. From where you left off I made one change here: https://github.com/mysociety/alaveteli/pull/4480/commits/e5f7a259816e659e311bde888bc6094a2d7ebedf but this doesn't seem to effect the rendering. Would you have any time to take a look?

FF (looks good): 
screen shot 2018-02-06 at 11 20 27

Safari (random list bullet): 
screen shot 2018-02-06 at 11 27 41

Chrome (larger white gaps) see Gareth's review above.

File: config/general.yml-example (hunk 1 above)
[the remark this answers is not in the document]
Comment (gbp) [reply config/general.yml-example]: Yes. I'll rename this.

**Comment by zarino** [comment]
@garethrees @gbp I’ve fixed the whitespace issue in Chrome and the bullet point issue in Safari, and also made the carets clickable, and made the carets turn into little loading spinners while the ajax request is in flight.

@gbp Do you think we need some error handling on the ajax handler in `fetchBodies()`, in case the request fails?

**Comment by garethrees** [comment]
Merged as https://github.com/mysociety/alaveteli/pull/4538

**Pull request closed without merging**